src/clients/python/CMakeLists.txt

```diff
@@ -33,6 +33,7 @@ add_library(crequest SHARED crequest.cc)
 target_link_libraries(
   crequest
   PUBLIC request
+  PRIVATE protobuf::libprotobuf
```

Comment:
Why is this needed?

Reply:
Otherwise there will be linker error: `unresolved external symbol "public: bool __cdecl google::protobuf::MessageLite::SerializeToString()`, this function is called [here](https://github.com/NVIDIA/tensorrt-inference-server/blob/master/src/clients/python/crequest.cc#L249).
But this error can also be resolved if the "request" library link gRPC as PUBLIC instead of [PRIVATE](https://github.com/NVIDIA/tensorrt-inference-server/pull/403/files#diff-a4364faad3475ef0a2269ab8f0f606acR48), perhaps gRPC uses protobuf and thus exposes the symbol?

Reply:
We don't want librequest to expose the protobuf symbols since that can cause problems when the main application is using their own protobuf... I see why we need it, but I wonder why we don't for linux?

Reply:
I am not sure, perhaps they use different strategy on exporting symbols? If I `nm librequest.so`, this symbol is included, and it is not the case on Windows (`dumpbin /EXPORTS request.dll`)

```
nm librequest.so | grep SerializeToString
000000000078eeec T _ZNK6google8protobuf11MessageLite17SerializeToStringEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
```

Reply:
I guess we should include an ldscript for librequest.so like we do for libtrtserver.so in src/servers/CMakeList.txt... but I'm a little hesitant because users will need our exact version of protobuf to use the library. Perhaps for consistency between linux and win32 we should just make libprotobuf public for librequest and then we don't need it explicitly in crequest.